lib/ExpensiMark.js

```diff
@@ -476,9 +470,28 @@ export default class ExpensiMark {
             }
             replacedText = replacedText.concat(textToCheck.substr(startIndex, (match.index - startIndex)));
 
+            // we want to avoid matching email address domains
+            let abort = false;
+            let shouldRetryByAtSign = false;
+
+            if ((match.index !== 0) && (textToCheck[match.index - 1] === '@')) {
+                const domainRegex = new RegExp('^(([a-z-0-9]+\\.)+[a-z]{2,})(\\S*)', 'i');
+                const domainMatch = domainRegex.exec(match[2]);
+
+                // e.g. test@expensify.com/https://www.test.com
+                // If the matched string faces @ sign before it,
+                // We will retry to apply autolink rule to the string(e.g. /https://www.test.com) except for domain(e.g. expensify.com) after @ sign.
```

Comment:
Can you update this comment to make it clearer?
- What is the e.g. an example of? shouldn't that be after another comment?
- What does "faces @ sign before it"? I suggest "If the matches string has a leading @ sign" -- use "leading" or "trailing" depending on the case
- It should say "why"

Reply:
> What is the e.g. an example of? shouldn't that be after another comment?

yes, it's example of comment. It should be after comment normally, but I tended to point '/https://www.test.com' and 'test@expensify.com' in comment details

> What does "faces @ sign before it"? I suggest "If the matches string has a leading @ sign" -- use "leading" or "trailing" depending on the case

Your suggestion is good, I will change comment as you recommended. "if the matched string has a leading @ sign"

> It should say "why"

By a Old PR, if matched string has leading @ sign, It simply aborts to apply autolink rule. But we have edge cases like test@expensify.com/https://www.test.com, where https://www.test.com should be parsed as url.
To explain above reason, I added a example of edge test string